src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -1147,13 +1148,127 @@ int set_committed_logging_object(RadosStore* store,
+  librados::IoCtx io_ctx;
+  if (const int ret = rgw_init_ioctx(dpp, store->getRados()->get_rados_handle(), data_pool, io_ctx); ret < 0) {
+    return -EIO;
+  }
+  if (!io_ctx.is_valid()) {
+    return -EIO;
+  }
+  bufferlist bl;
+  librados::ObjectReadOperation op;
+  int rval;
+  op.getxattr(RGW_ATTR_LOGGING_EC_POOL, &bl, &rval);
+  if (const int ret = rgw_rados_operate(dpp, io_ctx, obj_name_oid, std::move(op), nullptr, y); ret < 0) {
+    return ret;
+  }
+  ceph::decode(is_ec_pool, bl);
+  return 0;
+}
+
+int set_is_logging_bucket_ec_pool (RadosStore* store,
+    const DoutPrefixProvider *dpp,
+    const std::string& obj_name_oid,
+    const rgw_pool& data_pool,
+    bool& is_ec_pool,
+    optional_yield y) {
+
+  // Find out if the pool is erasure coded and set the attr
+  is_ec_pool = false;
+  librados::Rados* rados = store->getRados()->get_rados_handle();
+  auto ret = rados->mon_command(
+      "{\"prefix\": \"osd pool get\", \"pool\": \"" +
+      data_pool.name + "\", \"var\": \"erasure_code_profile\"}",
+      {}, NULL, NULL);
+  if (ret < 0 && ret != -EACCES) {
+    ldpp_dout(dpp, 10) << __func__ << " ERROR: failed to find out if pool"
+                       << data_pool.name << " is erasure coded. ret = "
+                       << ret << dendl;
+    return -EIO;
+  } else if (ret == 0){
+    is_ec_pool = true;
+  }
+  bufferlist bl;
+  ceph::encode(is_ec_pool, bl);
+
+  // Return success even if we fail post this as we have the is_ec_pool value
+  librados::IoCtx io_ctx;
+  if (const int ret = rgw_init_ioctx(dpp, store->getRados()->get_rados_handle(), data_pool, io_ctx); ret < 0) {
+    ldpp_dout(dpp, 10) << "ERROR: failed to get IO context when setting is_ec_pool attr on :" << obj_name_oid << dendl;
+    return 0;
+  }
+  if (!io_ctx.is_valid()) {
+    ldpp_dout(dpp, 10) << "ERROR: invalid IO context when setting is_ec_pool attr" << dendl;
+    return 0;
+  }
+
+  librados::ObjectWriteOperation op;
+  // if object does not exist, we should not create the attribute
+  op.assert_exists();
+  op.setxattr(RGW_ATTR_LOGGING_EC_POOL, std::move(bl));
+  if (const int ret = rgw_rados_operate(dpp, io_ctx, obj_name_oid, std::move(op), y); ret < 0){
+    ldpp_dout(dpp, 10) << "ERROR: failed to set is_ec_pool attr on :" << obj_name_oid
+                       << " ret: " << ret << dendl;
+  }
+  return 0;
+}
+
+
+int get_logging_temp_object_pool (const DoutPrefixProvider *dpp,
+    RadosBucket *bucket,
+    RadosStore* rados_store,
+    const std::string& obj_name_oid,
+    rgw_pool& data_pool, bool& is_ec_pool, optional_yield y) {
+
+  is_ec_pool = false;
+  rgw_obj obj {bucket->get_key(), "dummy"};
+  if (!rados_store->getRados()->get_obj_data_pool(bucket->get_placement_rule(), obj, &data_pool)) {
+    ldpp_dout(dpp, 1) << "ERROR: failed to get data pool for bucket '" << bucket->get_key() <<
+	"' when writing logging object" << dendl;
+    return -EIO;
+  }
+
+  auto ret = get_is_logging_ec_pool(rados_store, obj_name_oid,
```

Comment:
if the get/set functions are only used here we can probably create one function that check if the ec_pool flag exists, and if not, set it

Reply:
Done.